marlowe-playground-client/src/Marlowe/Gists.purs

```diff
@@ -39,45 +41,47 @@ type PlaygroundFiles
 
 toArray :: PlaygroundFiles -> Array NewGistFile
 toArray { playground, marlowe, haskell, blockly, javascript, actus } =
-  [ mkNewGistFile playgroundFilename playground
+  [ mkNewGistFile filenames.playground playground
   ]
     <> catMaybes
-        [ mkNewGistFile marloweFilename <$> marlowe
-        , mkNewGistFile haskellFilename <$> haskell
-        , mkNewGistFile blocklyFilename <<< unwrap <$> blockly
-        , mkNewGistFile jsFilename <$> javascript
-        , mkNewGistFile actusFilename <<< unwrap <$> actus
+        [ mkNewGistFile filenames.marlowe <$> marlowe
+        , mkNewGistFile filenames.haskell <$> haskell
+        , mkNewGistFile filenames.blockly <<< unwrap <$> blockly
+        , mkNewGistFile filenames.javascript <$> javascript
+        , mkNewGistFile filenames.actus <<< unwrap <$> actus
         ]
 
-playgroundFilename :: String
-playgroundFilename = "playground.marlowe.json"
-
-marloweFilename :: String
-marloweFilename = "playground.marlowe"
-
-haskellFilename :: String
-haskellFilename = "Main.hs"
-
-blocklyFilename :: String
-blocklyFilename = "blockly.xml"
-
-jsFilename :: String
-jsFilename = "playground.js"
-
-actusFilename :: String
-actusFilename = "actus.xml"
+filenames ::
+  { playground :: String
+  , marlowe :: String
+  , haskell :: String
+  , blockly :: String
+  , javascript :: String
+  , actus :: String
+  }
+filenames =
+  { playground: "playground.marlowe.json"
+  , marlowe: "playground.marlowe"
+  , haskell: "Main.hs"
+  , blockly: "blockly.xml"
+  , javascript: "playground.js"
+  , actus: "actus.xml"
+  }
```

Comment:
I'd not thought of this, but a PS extensible record is actually a nice way of doing a map with _some_ mandatory keys. 🤔

Reply:
💯